pennylane/pauli/pauli_arithmetic.py

```diff
@@ -143,6 +143,7 @@ def __init__(self, mapping):
         for wire, op in mapping.copy().items():
             if op == I:
                 del mapping[wire]
+        mapping = dict(sorted(mapping.items(), key=lambda item: str(item[0])))
```

Comment:
I am a little curious about this change. What's the benefit of doing this? Part of the reasoning for going with a dictionary was that dictionaries are inherently unordered, so I'm curious as to whether or not there is Python black magic going on.

Reply:
alas, they're kinda ordered: https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6

Reply:
So were we sensitive to the "kinda order" somehow?

Reply:
There was at least one test, `test_clifford` in `tests/qchem/test_tapering.py`, that was failing without that change. And the test fails because of the introduction of `term_2_pauli_word` in the Hamiltonian conversion routine. If that's preferable, we could also change the last test assert to `assert pauli_sentence(result_as_op) == pauli_sentence(u)` which will pass without that change.

Reply:
So basically going to pauli sentence and back changes the order slightly so comparison no longer works?

I think I'd prefer to change the test for now and remove this sorting here.  I think we don't want any additional overheads here.

We should potentially look into improving comparison for sums to make them less order-sensitive.

Reply:
Agreed. Let's see if any other test fails.